src/sharedHooks/useNumUnuploadedObservations.js

```diff
@@ -0,0 +1,14 @@
+// @flow
+
+import { RealmContext } from "providers/contexts";
+import Observation from "realmModels/Observation";
+
+const { useRealm } = RealmContext;
+
+const useNumUnuploadedObservations = ( ): Object => {
+  const realm = useRealm( );
+  const unsyncedObs = Observation.filterUnsyncedObservations( realm );
+  return unsyncedObs?.length;
```

Comment:
A call to `Realm.Collection.filtered()` as done in `filterUnsyncedObservations` will always return an array, even if no matching entries are found, so we can remove the optional chaining here. Then the return of this hook will always be a number. This would make for a clearer intent IMO.